File: android/app/src/main/java/com/theoplayer/contentprotectionintegration/TypeUtils.java
package com.theoplayer.contentprotectionintegration;

import org.json.JSONArray;
import org.json.JSONObject;

public class TypeUtils {
    /**
     * Convert an array of bytes to a JSONArray object.
     */
    public static JSONArray fromByteArrayToUint8JsonArray(final byte[] bytes) {
        JSONArray jsonArray = new JSONArray();
        for (byte aByte : bytes) {
            jsonArray.put(aByte & 0xff);
        }
        return jsonArray;
    }

    /**
     * Convert a JSON object to an array of bytes.
     */
    public static byte[] fromJsonToByteArray(final JSONObject jsonObject) {
        return jsonObject.toString().getBytes();
    }
}

File: android/app/src/main/java/com/theoplayer/contentprotectionintegration/TypeUtils.kt
package com.theoplayer.contentprotectionintegration

import org.json.JSONArray
import org.json.JSONObject

object TypeUtils {
    /**
     * Convert an array of bytes to a JSONArray object.
     */
    fun fromByteArrayToUint8JsonArray(bytes: ByteArray): JSONArray {
        val jsonArray = JSONArray()
        for (aByte in bytes) {
            jsonArray.put((aByte.toInt() and 0xff).toByte())
        }
        return jsonArray
    }

    /**
     * Convert a JSON object to an array of bytes.
     */
    fun fromJsonToByteArray(jsonObject: JSONObject): ByteArray {
        return jsonObject.toString().toByteArray()
    }
}
